sqlglot/dialects/doris.py

```diff
@@ -11,6 +13,54 @@
 from sqlglot.dialects.mysql import MySQL
 
 
+def _to_date_sql(
+    self: generator.Generator, expression: Union[exp.Anonymous, exp.TsOrDsToDate]
+) -> str:
+    if isinstance(expression, exp.Anonymous):
+        expressions = expression.args.get("expressions")
+        if expressions is None or len(expressions) == 0:
+            raise ValueError("Missing expressions for TO_DATE")
+        expr = expressions[0]
+        time_format = expressions[1] if len(expressions) > 1 else None
+        expr_TIMESTAMP = self.sql(expression, "this")
+        if expr_TIMESTAMP == "TIMESTAMP":
+            return f"TIMESTAMP({expr})"
+    elif isinstance(expression, exp.TsOrDsToDate):
+        expr = expression.args.get("this")
+        time_format = expression.args.get("format")
+    else:
+        raise ValueError("Invalid expression type")
+
+    if time_format is not None:
+        return f"STR_TO_DATE({expr}, {time_format})"
+    else:
+        return f"TO_DATE({expr})"
+
+
+def handle_nvl2(self, expression: exp.Nvl2) -> str:
+    this = self.sql(expression, "this")
+    expr1 = self.sql(expression, "true")
+    expr2 = self.sql(expression, "false")
+    return f"CASE WHEN {this} IS NOT NULL THEN {expr2} ELSE {expr1} END"
```

Comment:
Since there is no nvl2 function in doris, I do it this way

Reply:
i will change it

Reply:
Another suggestion, in case you want to make this change: instead of hard-coding the `CASE ...` as a string, it's better to create the equivalent `exp.Case` expression and then generate SQL for that, in order to leverage the formatting logic in `case_sql`. So, something like:

```python
def nvl2_sql(self, expression: exp.Nvl2) -> str:
    if self.NVL2_SUPPORTED:
        return self.func("NVL2", *expression.args.values())

    case = exp.Case().when(
        expression.this.is_(exp.null()).not_(copy=False), expression.args["true"].copy(), copy=False
    )
    else_cond = expression.args.get("false")
    if else_cond:
        case.else_(else_cond)

    return self.sql(case)
```

Reply:
ok，thanks